lift/Readme.md

```diff
@@ -6,271 +6,360 @@
-working in a team, it can be very useful to add a custom name to the migration. A name can be provided like this:
+**Note:** `migrate save` doesn't run migrations, it simply creates them.
 
-```bash
-$ prisma migrate --name my-initial-migration
+#### Up
+
+Up runs your outstanding migrations against the datasources, synchronizing the datasources with your schema file. Up depends on the migrations generated by the
+Save method.
+
+Up runs the unapplied `up` scripts in ascending order by the migration's timestamp. For example, if we have the following migrations directory:
+
+```
+migrations/
+  └─ 20190920142118-initial/
+    └─ steps.json
+    └─ schema.prisma
+    └─ README.md
+    └─ before.sh
+    └─ after.sh
+  └─ 20190920142120-add-user/
+    └─ steps.json
+    └─ schema.prisma
+    └─ README.md
+    └─ before.sh
+    └─ after.sh
 ```
 
-This will create the migration folder `migrate/20190322092247-my-initial-migration/` with the following content:
+The order of execution is the following:
 
-```bash
-.
-└── 20190322092247-init
-    ├── client.ts
-    └── datamodel.mdl
+1. `20190920142118-initial/before.sh`
+2. `20190920142118-initial/steps.json`
+3. `20190920142118-initial/after.sh`
+4. `20190920142120-add-user/before.sh`
+5. `20190920142120-add-user/steps.json`
+6. `20190920142120-add-user/after.sh`
+
+#### Down
+
+Down rolls back the migrations you've applied against your datasources.
+
+Down runs the `down` scripts in descending order by the migration's timestamp. For example, if we have the following migrations directory:
+
+```
+migrations/
+  └─ 20190920142118-initial/
+    └─ steps.json
+    └─ schema.prisma
+    └─ README.md
+    └─ before.sh
+    └─ after.sh
+  └─ 20190920142120-add-user/
+    └─ steps.json
+    └─ schema.prisma
+    └─ README.md
+    └─ before.sh
+    └─ after.sh
 ```
 
-This means that every migration will have its own generated Prisma Client in the chosen language (JS, TS or Go).
+The order of execution is the following:
 
-### Writing `before` or `after` migration scripts
+1. `20190920142120-add-user/after.sh`
+2. `20190920142120-add-user/steps.json`
+3. `20190920142120-add-user/before.sh`
+4. `20190920142118-initial/after.sh`
+5. `20190920142118-initial/steps.json`
+6. `20190920142118-initial/before.sh`
 
-There are a few use-cases when you want to run your own custom migration logic:
+### Lift Server
 
-- Making sure that the data adheres to constraints that you want to introduce, like `unique` or `non null` constraints
-- Adding specific database primitives like functions in Postgres, which are not yet supported by Prisma
-- Initializing data after a migration has been executed. Both SQL or the Prisma Client could be useful here
+The Lift server is a low-level interface that the Lift Client communicates with. Currently the client communicates to the migration server in the JSONRPC format
```

Comment:
migration server = lift server?

Reply:
Should we actually call those servers? Or should we rather embrace our internal engine term?